File: Data Structure/Array or Matrix/LET_no735_Asteroid Collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        result = []
        for num in asteroids:
            if num > 0:
                result.append(num)
            else:
                while result and result[-1] > 0 and result[-1] < -num:
                    result.pop()
                if not result or result[-1] < 0:
                    result.append(num)
                elif result[-1] == -num:
                    result.pop()
        return result

File: Data Structure/Array or Matrix/LET_no735_Asteroid Collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
        vector<int> res;
        for (int ast : asteroids) {
            if (ast > 0)
                res.push_back(ast);
            else {
                while (!res.empty() && res.back() > 0 && res.back() < -ast)
                    res.pop_back();

                if (res.empty() || res.back() < 0)
                    res.push_back(ast);
                else if (res.back() == -ast)
                    res.pop_back();
            }
        }

        return res;
    }
};